lib/utils.ts

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable @typescript-eslint/prefer-for-of */
+import type {OnyxKey} from './types';
+
+type EmptyObject = Record<string, never>;
+type EmptyValue = EmptyObject | null | undefined;
+
+function isEmptyObject<T>(obj: T | EmptyValue): obj is EmptyValue {
+    return typeof obj === 'object' && Object.keys(obj || {}).length === 0;
+}
+
+// Mostly copied from https://medium.com/@lubaka.a/how-to-remove-lodash-performance-improvement-b306669ad0e1
+
+function isMergeableObject(value: unknown): boolean {
```

Comment:
Can you explain for this function in plain English when the object is mergeable?